src/protocols/handshake.rs

```diff
@@ -115,20 +127,72 @@ impl Protocol for HandshakeProtocol {
     }
 
     fn receive_data(&mut self, data: Vec<u8>) {
-        if data.len() != 8 {
-            // some payload error
-            let cbor_value: Value = de::from_slice(&data[..]).unwrap();
-            match self.find_error_message(&cbor_value) {
-                Ok(error_message) => {
-                    self.result = Some(Err(error_message));
-                }
-                Err(_) => {
-                    self.result = Some(Err(format!("Unable to parse payload error! {}", hex::encode(data))));
+        let confirm: Value = de::from_slice(&data[..]).unwrap();
```

Comment:
We need to first turn this into a separate _output validation_ function. Then the verification function should work as a sequence of checks in the form of _error handling_ rather than _nested matching_.

Reply:
fixed. Please take a look